waspc/cli/src/Wasp/Cli/Command/BuildStart/Config.hs

```diff
@@ -0,0 +1,49 @@
+module Wasp.Cli.Command.BuildStart.Config
+  ( BuildStartConfig (..),
+    clientPort,
+    clientUrl,
+    dockerContainerName,
+    dockerImageName,
+    serverUrl,
+  )
+where
+
+import Data.Char (toLower)
+import qualified StrongPath as SP
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec.Valid as ASV
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.ServerGenerator.Common (defaultDevServerUrl)
+import Wasp.Generator.WebAppGenerator.Common (defaultClientPort, getDefaultDevClientUrl)
+import Wasp.Project.Common (WaspProjectDir, makeAppUniqueId)
+
+data BuildStartConfig = BuildStartConfig
+  { appSpec :: AppSpec,
+    projectDir :: SP.Path' SP.Abs (SP.Dir WaspProjectDir),
+    buildDir :: SP.Path' SP.Abs (SP.Dir ProjectRootDir)
+  }
+
+-- NOTE(carlos): For now, creating these URLs and ports uses the default values
+-- we've hardcoded in the generator. In the future, we might want to make these
+-- configurable via the Wasp app spec or command line arguments.
+
+clientUrl :: BuildStartConfig -> String
```

Comment:
I would add a comment here saying that this implementation assumes that `clientPort` below is returning `defaultClientPort`, and if that changes, this also needs to update.

I am even tempted to merge these two into one function, called `clientPortAndUrl :: BuildStartConfig -> (Int, String)` as that would make it harder to mess this up, makes connection stronger.

Reply:
Done.
TBH what I'd do in JS (and would like to do here) would be something like:

```js
const clientPort = new URL(clientURL).port
```

Makes much more sense. Is there a way to do it in Haskell?

Reply:
Well I suggested in the previous comment somewhere that we could do the opposite, which is consturct url from port :D. I don't think we need to parse anything then.

And it could be just `buildClientUrl :: ClientPort -> ClientUrl`. Making up types now, maybe those are string or ints or whatever I don't know if we are branding them or not.